src/main/resources/org/jenkinsci/plugins/docker/workflow/Docker.groovy

```diff
@@ -81,10 +81,16 @@ class Docker implements Serializable {
         }
     }
 
+    String asEnv(String var) {
+        node {
+            script.isUnix() ? "$${var}" : "%${var}%"
```

Comment:
Isn't that wrong? This should cause the string interpolation of groovy to fail, shouldn't it?

I think this is correct:
```suggestion
            script.isUnix() ? "\$${var}" : "%${var}%"
```

Also with this change every command now enters a node and checks for unix two times. Once to find out what shell to use and once for the syntax of the variables. Is this an issue?

Reply:
Nor sure about interpolation, but we don't get any issue in my ide.

I'm also no Jenkins nor java expert, so i tried to be as less inversive as possible.

Please advise me for better solution to not have the duplicate node requests.

Reply:
let's fix the only the regession first and do other refactorings later. I'm happy to help.